File: ShippingMark/Models/ChiTietHangHoa.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Collections.Generic;
namespace ShippingMark.Models
{   
    public class ChiTietHangHoa
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int ID { get; set; }       
        [Display(Name = "MODEL")]
        public string MODEL { get; set; }
        [Display(Name = "SIZE Kích thước")]
        public string SIZE { get; set; }
        [Display(Name = "Số lượng")]
        public int SoLuongThamKhao { get; set; }
        // Thêm các trường còn lại theo cấu trúc bảng IFT
        [Display(Name = "Net weight")]
        public string NW { get; set; }
        [Display(Name = "Gross weight")]
        public string GW { get; set; }        
        [Display(Name = "Dài (m)")]
        public decimal? Dai { get; set; }

        [Display(Name = "Rộng (m)")]
        public decimal? Rong { get; set; }
        [Display(Name = "Cao (m)")]
        public decimal? Cao { get; set; }
        
        [Display(Name = "Màu (Color)")]
        public string COLOR { get; set; }
        [Display(Name = "Loại đóng gói")]
        public TypeNum SelectedTypeNum { get; set; }

        [Display(Name = "Nội dung mở rộng")]
        public string ExtendedContent { get; set; }
        [Display(Name = "In")]
        public bool? UsingPrint { get; set; }      
        [Display(Name = "Lô")]
        public string LotNo { get; set; }
        public string QUANTITY { get; set; }
        [Display(Name = "Thông báo xuất hàng")]
        public string CreatedBy { get; set; }
        [Column(TypeName = "datetime2")]
        public DateTime CreatedDate { get; set; }
        public string LastModifiedBy { get; set; }
        [Column(TypeName = "datetime2")]
        public DateTime? LastModifiedDate { get; set; }
        [Display(Name = "Thông báo xuất hàng")]
        [ForeignKey("ThongBaoXuatHang")]
        public int ThongBaoXuatHang_ID { get; set; }
        public virtual ThongBaoXuatHang ThongBaoXuatHang { get; set; }
        // === BỔ SUNG LIÊN KẾT NGƯỢC VỀ CONTAINER (QUAN HỆ NHIỀU-NHIỀU) ===
        public virtual ICollection<Container> Containers { get; set; }
    }
}

File: app/Models/ChiTietHangHoa.php
<?php

namespace App\Models;

use App\Enums\TypeNum;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Spatie\Activitylog\Models\Concerns\LogsActivity;
use Spatie\Activitylog\Support\LogOptions;

class ChiTietHangHoa extends Model
{
    use HasFactory, LogsActivity;

    protected $table = 'chi_tiet_hang_hoas';

    protected $fillable = [
        'thong_bao_xuat_hang_id',
        'model',
        'size',
        'so_luong_tham_khao',
        'nw',
        'gw',
        'dai',
        'rong',
        'cao',
        'color',
        'selected_type_num',
        'extended_content',
        'using_print',
        'lot_no',
        'quantity',
        'created_by',
        'last_modified_by',
    ];

    protected $casts = [
        'selected_type_num' => TypeNum::class,
        'using_print' => 'boolean',
        'so_luong_tham_khao' => 'integer',
        'dai' => 'decimal:3',
        'rong' => 'decimal:3',
        'cao' => 'decimal:3',
    ];

    public function getActivitylogOptions(): LogOptions
    {
        return LogOptions::defaults()
            ->logFillable()
            ->logOnlyDirty()
            ->dontLogEmptyChanges();
    }

    public function thongBaoXuatHang()
    {
        return $this->belongsTo(ThongBaoXuatHang::class, 'thong_bao_xuat_hang_id');
    }

    public function containers()
    {
        return $this->hasMany(Container::class, 'chi_tiet_hang_hoa_id');
    }
}
